modules/mono/glue/cs_files/NodeExtensions.cs

```diff
@@ -0,0 +1,10 @@
+namespace Godot
+{
+    public partial class Node
+    {
+        public T GetNode<T>(NodePath path) where T : Godot.Node
+        {
+            return (T)GetNode(path);
```

Comment:
Why not `return GetNode(path) as T;`? The `as` keyword will return null if the cast is invalid, rather than throwing [InvalidCastException](https://msdn.microsoft.com/en-us/library/system.invalidcastexception).

Reply:
Toward the end of the linked issue, it seems as though @reduz, @neikeq and @mysticfall (and myself) all agree not to allow it to throw an exception, which doing a direct cast will do if the type doesn't match. Thus it should be `GetNode(path) as T`, but maybe I'm misunderstanding them as earlier there was discussion saying it should throw an exception on a mismatched type. So should it throw an exception on a mismatched type, but not if the node doesn't exist, or should it always return null instead of any exception?

If the consensus was only about not throwing an exception if it wasn't found, then the direct cast should stay. If it should never throw an exception, then is should be `as T`.

Reply:
> if GetNode returned null, a direct cast would always cause an exception

Are you sure about that?

Reply:
First off, C# allows you to cast null to anything, so this method does not throw if the node does not exist. We're fine there.

I can see that the thread reached consensus not to throw on null, but I *can't* see a clear consensus on "what to do if you've got the wrong type"

Reply:
Like I commented on the issue, I'm all for it to throw InvalidCastException. My negative comments were against NodeNotFoundException and IndexOutOfRangeException, because that would make people assume the auto-generated non-generic version of those methods also threw them, which they don't.
I also commented why I'm against using `as`, a.k.a. returning null on invalid cast instead of throwing.

Reply:
Ah, okay, it did get a little confusing. InvalidCastException - Fine, NullReferenceException - Bad (because default method doesn't do it).